doc/media-info-enhancements/media-info.md

```diff
@@ -0,0 +1,170 @@
+### Goals
+The goal is to have xcvrd build a Python dictionary, containing all static transceiver information (i.e. transceiver information that does not change over time) for an installed transceiver, information that is needed to:
+- properly configure the NPU for the installed transceiver, e.g. set the NPU interface type to "fiber" or "copper";
+- provide information for making a platform-dependent, port-dependent and transceiver-dependent lookup of port configuration information to be passed to the NPU, e.g. look up port pre-emphasis value(s);
+- provide information for determining possible port configurations, and validating requested port configurations, e.g. determining what, if any, breakout modes are possible on a port; and
+- display transceiver information to the user.
+
+The format of this dictionary, i.e. the member fields, and the contents of all those fields, shall be standardized, and platform-independent, so that the code to configure the NPU and display information to the user can also be platform-independent.
+
+This dictionary, once created by xcvrd, is published to the Redis DB, as a record in the TRANSCEIVER_INFO table.  NPU-configuration and CLI code will obtain transceiver information from there.
+### Current format
+An example of the transceiver information dictionary constructed by xcvrd is as follows:
+~~~
+{
+	"Connector": "No separable connector", 
+	"cable_length": "2", 
+	"cable_type": "Length Cable Assembly(m)", 
+	"encoding": "64B66B", 
+	"ext_identifier": "Power Class 1(1.5W max)", 
+	"ext_rateselect_compliance": "QSFP+ Rate Select Version 1", 
+	"hardwarerev": "A0", 
+	"manufacturename": "DELL", 
+	"modelname": "76V43", 
+	"nominal_bit_rate": "255", 
+	"serialnum": "CN0LXD0096H3A36", 
+	"specification_compliance": "{'10/40G Ethernet Compliance Code': '40GBASE-CR4'}", 
+	"type": "QSFP28 or later", 
+	"type_abbrv_name": "QSFP28", 
+	"vendor_date": "2019-06-17 ", 
+	"vendor_oui": "3c-18-a0"
+}
+~~~
+### Proposed format
+The intention is to maintain existing field names and their contents, for backward-compatibility.  New fields shall be added.
+
+The following shall be added to the dictionary:
+
+<h4>cable_breakout</h4>
+<table style="border-style:none;">
+	<tr>
+		<td valign="top">Description</td>
+		<td>Breakout of attached cable, if any, i.e. the number of cables directly attached to transceiver</td>
+	</tr>
+	<tr>
+		<td valign="top">Value(s)</td>
+		<td>integer; 0 when no directly-attached cable</td>
+	</tr>
+	<tr>
+		<td valign="top">Needed by</td>
+		<td>Validation of port breakout configurations, i.e. whether or not transceiver supports breakout</td>
+	</tr>
+</table>
+<h4>cable_length_detailed</h4>
+<table style="border-style:none;">
+	<tr>
+		<td valign="top">Description</td>
+		<td>Length of directly-attached cable, in metres</td>
```

Comment:
i can see the cable_length already in the schema, what is the difference here?

Reply:
Please refer to the section "Notes" below.
The existing "cable_length" entry is an integer, and so cannot represent lengths that are not whole numbers.

Reply:
From the specification(SFF8472, 8436,...) perspective, shall the cable length be an integer or float? If it shall be a float then this should be fixed in the parsing code?